src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/CalculateStakeRule.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.Consensus.Rules.CommonRules
+{
+    /// <summary>
+    /// Checks if <see cref="Block"/> has a valid PoS header.
+    /// </summary>
+    public class CalculateStakeRule : ConsensusRule
+    {
+        /// <summary>Quick easy access to the POS rules.</summary>
+        private PosConsensusRules posParent;
```

Comment:
Make this readonly?

Reply:
yes

Reply:
Ah this can be readonly it initialised out of the constructor

Reply:
I am considering now to make a POSBaseRule that will hold the PosParent.